ramalama/common.py

```diff
@@ -165,25 +186,67 @@ def download_file(url, dest_path, headers=None, show_progress=True):
         headers (dict): Optional headers to include in the request.
         show_progress (bool): Whether to show a progress bar during download.
 
-    Returns:
-        None
+    Raises:
+        RuntimeError: If the download fails after multiple attempts.
     """
-    http_client = HttpClient()
-
     headers = headers or {}
 
-    # if we are not a tty, don't show progress, can pollute CI output and such
+    # If not running in a TTY, disable progress to prevent CI pollution
     if not sys.stdout.isatty():
         show_progress = False
 
-    try:
-        http_client.init(url=url, headers=headers, output_file=dest_path, progress=show_progress)
-    except urllib.error.HTTPError as e:
-        if e.code == 416:  # Range not satisfiable
-            if show_progress:
-                print(f"File {url} already fully downloaded.")
-        else:
-            raise e
+    http_client = HttpClient()
+    max_retries = 5  # Stop after 5 failures
+    retries = 0
+
+    while retries < max_retries:
+        try:
+            # Initialize HTTP client for the request
+            http_client.init(url=url, headers=headers, output_file=dest_path, progress=show_progress)
+            logging.info(f"✅ Successfully downloaded {url} to {dest_path}.")
+            return  # Exit function if successful
+
+        except urllib.error.HTTPError as e:
```

Comment:
I think the fail messages are fine, but I would delete all the successful messages I think.

Reply:
sure